src/config.rs

```diff
@@ -5,13 +9,65 @@ pub trait Config {
 
     fn current_shell_name(&self) -> String;
 
-    fn set_current_shell_name(&mut self, name: &str);
+    fn set_current_shell_name(&mut self, name: &str) -> io::Result<()>;
 
-    fn does_shell_exist(&self, name: &str) -> bool;
+    fn does_shell_exist(&self, name: &str) -> io::Result<bool>;
+}
+
+#[derive(Clone)]
+pub struct FsConfig {
+    pub root_path: PathBuf,
+    pub current_shell: String,
+}
+
+impl FsConfig {
+    fn new(root_path: PathBuf) -> io::Result<Self> {
+        let config_path = root_path.join("current_shell");
```

Comment:
I'm a tad uncomfortable that this `current_shell` filename knowledge is duplicated here and on line 50.  Maybe pull out an `impl FsConfig` method for getting that `join`?